src/app/admin/page.tsx

```diff
@@ -1,60 +1,172 @@
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
   const router = useRouter();
-  
-  useEffect(() => {
-    const fetchUser = async () => {
-      try {
-        const fetchedUser = await getUser();
-        setUser(fetchedUser);
-      } catch (error) {
-        console.error("Error fetching user:", error);
-      } finally {
-        setIsLoading(false);
-      }
-    };
-
-    fetchUser();
-  }, []);
-
-  if (isLoading) {
-    return <div className="flex min-h-screen items-center justify-center text-3xl">Loading...</div>;
-  }
 
-  if (!user || !user.admin) {
+  // Filter AP classes based on search term
+  const filteredClasses = apClasses.filter((apClass) =>
+    apClass.toLowerCase().includes(searchTermAPClasses.toLowerCase()),
+  );
+
+  const { users, error, handleRoleChange } = useUserManagement(user);
+
+  const openModal = (user: User) => {
+    setSelectedUser(user);
+    setIsModalOpen(true);
+  };
+
+  const closeModal = () => {
+    setIsModalOpen(false);
+    setSelectedUser(null);
+  };
+
+  if (!user || user.access === "user") {
     router.push("/");
-  } else {
-    return (
-      <div>
-        <Navbar />
-
-        <div className="mx-auto mt-12 flex max-w-6xl flex-col px-8 pb-8 ">
-          <div className="mb-6 flex flex-col gap-1">
-            <h1 className="text-balance text-left text-5xl font-extrabold lg:text-6xl">
-              Admin dashboard
-            </h1>
-            <p className="w-full text-pretty text-lg opacity-70 lg:text-xl">
-              Click the subject you wish to add articles to
-            </p>
-          </div>
-          <div className="grid grid-cols-3 gap-4 lg:gap-8">
-            <APLibrary />
+    return null;
+  }
+
+  return (
+    <div>
+      <Navbar />
+
+      <div className="mx-auto mt-12 flex max-w-6xl flex-col px-8 pb-8 ">
+        <div className="mb-6 flex flex-col gap-1">
+          <h1 className="text-balance text-left text-5xl font-extrabold lg:text-6xl">
+            Admin dashboard
+          </h1>
+
+          {user.access === "admin" && (
+            <>
+              <p className="w-full text-pretty text-lg opacity-70 sm:text-lg lg:text-xl">
+                Change the role of a user
+              </p>
+              <div className="my-4 rounded-2xl border-2 p-4">
+                <div className="rounded-md bg-white p-4">
+                  <input
+                    type="text"
+                    className="mb-3 w-full rounded-md border p-2"
+                    placeholder="Search for a user..."
+                    value={searchTermUsers}
+                    onChange={(e) =>
+                      setSearchTermUsers(e.target.value)
+                    }
+                  />
+                  <ul className="class-list max-h-60 overflow-y-auto">
+                    {error && users!.map((u) => ( // If error, it will show error message. Otherwise, it will show users
```

Comment:
This is still so confusing to me, if error is falsy, doesn't that mean that users!.map... will show?

Reply:
I messed up lmfao I changed it dw